pypact/input/inputdata.py

```diff
@@ -458,3 +468,44 @@ def _deserialize(self, f):
                 if len(parts) != 2:
                     raise PypactInvalidOptionException("Invalid DENSITY line format.")
                 self.setDensity(float(parts[1]))
+
+            elif line.startswith("FLUX") and irradiation_active:
+                # Parse the FLUX line
+                parts = line.split()
+                if len(parts) != 2:
+                    raise PypactInvalidOptionException("Invalid FLUX line format.")
+                fluxAmp = float(parts[1])
+
+                # Ensure FLUX is set to a non-zero value before the first irradiation step
+                if not flux_set and fluxAmp <= 0.0:
+                    raise PypactInvalidOptionException("FLUX must be set to a non-zero positive value before the first irradiation step.")
+                flux_set = True
+
+                # If FLUX is 0.0, skip TIME parsing
+                if fluxAmp == 0.0:
+                    self.addIrradiation(0.0, fluxAmp)
+                    continue
+
+                # Look ahead for the TIME line
+                next_line = next(lines, None)
+                if next_line and next_line.strip().startswith("TIME"):
+                    time_parts = next_line.strip().split()
+                    if len(time_parts) != 3:
```

Comment:
I am not sure if it's as simple as that.

I need to re-read the documentation but I think you can have situations like:

```
TIME 0.1 YEARS ATOMS
```

and 

```
TIME 60
```
and even when the ATOMS is on the next line, such as

```
TIME 0.1
ATOMS
```